BDArmory/Control/MissileFire.cs

```diff
@@ -5635,7 +5641,8 @@ bool CheckEngagementEnvelope(IBDWeapon weaponCandidate, float distanceToTarget)
             if (!engageableWeapon.engageEnabled) return true;
             //if (distanceToTarget < engageableWeapon.GetEngagementRangeMin()) return false; //covered in weapon select logic
             //if (distanceToTarget > engageableWeapon.GetEngagementRangeMax()) return false;
-            if (distanceToTarget > (engageableWeapon.GetEngagementRangeMax() * 1.2f)) return false; //have Ai begin to preemptively lead target, instead of frantically doing so after weapon in range
+            //if (distanceToTarget > (engageableWeapon.GetEngagementRangeMax() * 1.2f)) return false; //have Ai begin to preemptively lead target, instead of frantically doing so after weapon in range
+            if (distanceToTarget > (engageableWeapon.GetEngagementRangeMax() + (vessel.Velocity().magnitude * 2))) return false; //have AI preemptively begin to lead 2s out from max weapon range
```

Comment:
You can probably just use the `(float)vessel.speed` or `(float)vessel.srfSpeed` value here instead of computing the magnitude of the velocity.